packages/react-native-performance/src/performance-entry.ts

```diff
@@ -0,0 +1,216 @@
+type MarkOptions = {
+  startTime: number;
+  detail?: string;
+}
+
+type MetricOptions = {
+  startTime: number;
+  value: string | number;
+  detail?: string;
+}
+
+type MeasureOptions = {
+  startTime: number;
+  detail: string;
+  duration?: number;
+}
+
+type TimingOptions = {
+  initiatorType?: string;
+  responseEnd?: number;
+  fetchStart?: number;
+  transferSize?: number;
+  connectEnd?: number;
+  connectStart?: number;
+  decodedBodySize?: number;
+  domainLookupEnd?: number;
+  domainLookupStart?: number;
+  encodedBodySize?: number;
+  redirectEnd?: number;
+  redirectStart?: number;
+  requestStart?: number;
+  responseStart?: number;
+  secureConnectionStart?: number;
+  serverTiming?: number[];
+  workerStart?: number;
+  workerTiming?: number[];
+}
+
+export type EntryType =
+  'mark' |
+  'measure' |
+  'resource' |
+  'metric' |
+  'react-native-mark';
+
+export class PerformanceEntry {
+  name: string;
+  entryType: EntryType;
+  startTime: number;
+  duration: number;
+
+  constructor(name: string, entryType: EntryType, startTime: number, duration: number) {
+    this.name = name;
+    this.entryType = entryType;
+    this.startTime = startTime;
+    this.duration = duration;
+  }
+
+  toJSON() {
+    return {
+      name: this.name,
+      entryType: this.entryType,
+      startTime: this.startTime,
+      duration: this.duration,
+    };
+  }
+}
+
+export class PerformanceMark extends PerformanceEntry {
+  detail?: string;
+
+  constructor(markName: string, markOptions: MarkOptions) {
+    super(markName, 'mark', markOptions.startTime, 0);
+    this.detail = markOptions.detail;
+  }
+
+  toJSON() {
+    return {
+      name: this.name,
+      entryType: this.entryType,
+      startTime: this.startTime,
+      duration: this.duration,
+      detail: this.detail,
+    };
+  }
+}
+
+export class PerformanceReactNativeMark extends PerformanceEntry {
+  constructor(name: string, startTime: number) {
+    super(name, 'react-native-mark', startTime, 0);
+  }
+}
+
+export class PerformanceMetric extends PerformanceEntry {
+  value: string | number;
```

Comment:
I'm wondering if this should allow more types or even be restricted to only number.

Reply:
Thing coming to my mind some size related, time related and count related metrics. I think it can be restricted to number. What do you think?